crates/rustc_codegen_spirv/src/lib.rs

```diff
@@ -142,7 +142,8 @@ fn is_blocklisted_fn<'tcx>(
                 Some(assoc) if assoc.ident(tcx).name == sym::fmt => match assoc.container {
                     ty::ImplContainer => {
                         let impl_def_id = assoc.container_id(tcx);
-                        tcx.impl_trait_ref(impl_def_id).map(|tr| tr.def_id)
+                        tcx.impl_trait_ref(impl_def_id)
+                            .map(|tr| tr.subst_identity().def_id)
```

Comment:
`subst_identity` is not necessarily cheap, since it replaces one encoding of generics with another (the name seems misleading, since "identity substitutions" should, well, produce the same result, while this doesn't).

A `.def_id()` method should exist on this `EarlyBinder<TraitRef>` type just like one exists on `Binder<TraitRef>`, but sadly it seems to be lacking. Instead, you can do this for now:

```suggestion
                        tcx.impl_trait_ref(impl_def_id)
                            .map(|tr| tr.skip_binder().def_id)
```
(`skip_binder` is "bad" because it can lead to mistakes by ignoring important contextual information, but `.def_id` is just an ID, not something in the typesystem that cares about that context)

Reply:
I mainly got the pattern from the respective PR where they made similar changes. I'm still not entirely sure what EarlyBinder is supposed to do exactly.